src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2302,6 +2302,56 @@ void processContextElement
 
   auto_ptr<DBClientCursor> cursor;
 
+  // This block is to avoids that several entities with the same ID gets updated at the same time, which is
+  // not allowed in NGSIv2. The if is needed, as multiple update has been allowed in NGSIv1 (maybe without
+  // thinking too much about it, but NGSIv1 behaviour now will break backward compatibility)
+  if (apiVersion == "v2")
+  {
+    int entitiesNumber;
+
+    try
+    {
+      LM_T(LmtMongo, ("count() in '%s' collection: '%s'",
+                      getEntitiesCollectionName(tenant).c_str(),
+                      query.toString().c_str()));
+
+      connection = getMongoConnection();
+      entitiesNumber = connection->count(getEntitiesCollectionName(tenant).c_str(), query);
+      releaseMongoConnection(connection);
+
+      LM_I(("Database Operation Successful (%s)", query.toString().c_str()));
+    }
```

Comment:
Fixed in bda9795.

Note I'm using a "Fatal Error" for that, given that if getMongoConnection() returns NULL something really bad is happening in the DB pool management logic (I have event though if that should cause the broker end, with LM_X).